pymatgen/electronic_structure/bandstructure.py

```diff
@@ -541,43 +605,59 @@ def get_kpoint_degeneracy(self, kpoint, cartesian=False, tol=1e-2):
-             "kpoints": []}
+        d = {
+            "@module": self.__class__.__module__,
+            "@class": self.__class__.__name__,
+            "lattice_rec": self.lattice_rec.as_dict(),
+            "efermi": self.efermi,
+            "kpoints": [],
+        }
         # kpoints are not kpoint objects dicts but are frac coords (this makes
         # the dict smaller and avoids the repetition of the lattice
         for k in self.kpoints:
             d["kpoints"].append(k.as_dict()["fcoords"])
-        d["bands"] = {str(int(spin)): self.bands[spin]
-                      for spin in self.bands}
+
+        d["bands"] = {str(int(spin)): self.bands[spin].tolist() for spin in self.bands}
         d["is_metal"] = self.is_metal()
         vbm = self.get_vbm()
-        d["vbm"] = {"energy": vbm["energy"],
-                    "kpoint_index": vbm["kpoint_index"],
-                    "band_index": {str(int(spin)): vbm["band_index"][spin]
-                                   for spin in vbm["band_index"]},
-                    'projections': {str(spin): v.tolist() for spin, v in vbm[
-                        'projections'].items()}}
+        d["vbm"] = {
+            "energy": vbm["energy"],
+            "kpoint_index": vbm["kpoint_index"],
+            "band_index": {
+                str(int(spin)): vbm["band_index"][spin] for spin in vbm["band_index"]
+            },
+            "projections": {
+                str(spin): v.tolist() for spin, v in vbm["projections"].items()
+            },
+        }
         cbm = self.get_cbm()
-        d['cbm'] = {'energy': cbm['energy'],
-                    'kpoint_index': cbm['kpoint_index'],
-                    'band_index': {str(int(spin)): cbm['band_index'][spin]
-                                   for spin in cbm['band_index']},
-                    'projections': {str(spin): v.tolist() for spin, v in cbm[
-                        'projections'].items()}}
-        d['band_gap'] = self.get_band_gap()
-        d['labels_dict'] = {}
-        d['is_spin_polarized'] = self.is_spin_polarized
+        d["cbm"] = {
+            "energy": cbm["energy"],
+            "kpoint_index": cbm["kpoint_index"],
+            "band_index": {
+                str(int(spin)): cbm["band_index"][spin] for spin in cbm["band_index"]
+            },
+            "projections": {
+                str(spin): v.tolist() for spin, v in cbm["projections"].items()
+            },
+        }
+        d["band_gap"] = self.get_band_gap()
+        d["labels_dict"] = {}
+        d["is_spin_polarized"] = self.is_spin_polarized
+
+        # MongoDB does not accept keys starting with $. Add a blanck space to fix the problem
         for c in self.labels_dict:
-            d['labels_dict'][c] = self.labels_dict[c].as_dict()['fcoords']
-        d['projections'] = {}
+            mongo_key = c if not c.startswith("$") else " " + c
+            d["labels_dict"][mongo_key] = self.labels_dict[c].as_dict()["fcoords"]
```

Comment:
This mongo key issue; would this be better handed in `monty` rather than an ad-hoc basis here?

Reply:
well its an issue that pertains to MongoDB specifically, and even MSONable is not really geared for MongoDB. I do dislike the ad-hoc fix, but I dislike it because its mutating data rather than just using a structure that is better. Why do `labels` have to be a dictionary? Why not be a list of `Tuple`s ?

Reply:
I think it's so you can look up a label based on fraction coord, no? It's just for efficiency's sake when finding the label you want

Reply:
That can be done internally and not be a limited on the data model.

Reply:
Having the labels and kpoints as a dict is very handy for a user. I can agree though that internally they do not need to be stored in a dict. But this would need to change the whole data stored, if I'm nor wrong...